PublicAPI/Microsoft.ApplicationInsights.dll/net45/PublicAPI.Shipped.txt

```diff
@@ -99,6 +99,11 @@ Microsoft.ApplicationInsights.DataContracts.ExceptionHandledAt
 Microsoft.ApplicationInsights.DataContracts.ExceptionHandledAt.Platform = 2 -> Microsoft.ApplicationInsights.DataContracts.ExceptionHandledAt
 Microsoft.ApplicationInsights.DataContracts.ExceptionHandledAt.Unhandled = 0 -> Microsoft.ApplicationInsights.DataContracts.ExceptionHandledAt
 Microsoft.ApplicationInsights.DataContracts.ExceptionHandledAt.UserCode = 1 -> Microsoft.ApplicationInsights.DataContracts.ExceptionHandledAt
+Microsoft.ApplicationInsights.DataContracts.ExceptionDetailsInfo
```

Comment:
Should this collection define it's own collection inside to indicate parent-child relationship of exceptions?

Reply:
If we want to indicate parent-child relationship, I don't think there is any need to have own collection, but what we really want is to expose `id` and `outerId` fields of `ExceptionDetails`. Adding just `getters` for them doesn't add value, because user don't know how those ids were populated. If we add `setters` as well, then we will have to add extra logic to sync-up the relation reflected in `ExceptionDetailsInfo` into `ExceptionDetails` and vice-versa. I was not planning to do that (at least not in this PR).

 The problem that I was hoping to solve is if you have `System.Exception` and AI inner code flattens it, let us give opportunity to user to modify message name and type name before sending it over the wire.  @dnelly  and my scenario falls under this category. Alternatively, user can do it in his code, walks the exception objects tree and populate the flatten list of `ExceptionDetailsInfo` and just sets in `ExceptionTelemetry`. But I guess, AI already has that code, so opted for former option.

---
In reply to: [203866000](https://github.com/Microsoft/ApplicationInsights-dotnet/pull/865#discussion_r203866000) [](ancestors = 203866000)